internal/js/modules/k6/browser/common/selectors.go

```diff
@@ -71,107 +71,86 @@ func (s *Selector) appendPart(p *SelectorPart, capture bool) error {
+
 		eqIndex := strings.Index(part, "=")
 		var name, body string
 
 		switch {
-		// Using TrimQuote prevents issues with selectors, allowing us to
-		// handle quoted text with internal selectors, such as:
-		//
-		//       page.getByRole("listitem >> internal:has-text='Product 2'")
-		//
-		// We'd receive the following error because of the quotes:
-		//
-		//       Uncaught (in promise) getting text content of
-		//       "internal:role=listitem >> internal:has-text='Product 2'
-		//       >> nth=0": SyntaxError: Unexpected token ''', "'Product 2'"
-		//       is not valid JSON
-		//
-		// Selectors, such as, internal:has-text, by their nature, are
-		// exposed to users, even if we don't document their use.
-		case strings.HasPrefix(part, "internal:has-text="):
-			name = "internal:has-text"
-			body = TrimQuotes(part[eqIndex+1:])
-		case strings.HasPrefix(part, "internal:has-not-text="):
-			name = "internal:has-not-text"
-			body = TrimQuotes(part[eqIndex+1:])
-		case strings.HasPrefix(part, "nth="):
-			name = "nth"
-			body = part[eqIndex+1:]
-		case eqIndex != -1 && reQueryEngine.Match([]byte(strings.TrimSpace(part[0:eqIndex]))):
-			name = strings.TrimSpace(part[0:eqIndex])
+		case eqIndex != -1 && reQueryEngine.MatchString(strings.TrimSpace(part[:eqIndex])):
+			name = strings.TrimSpace(part[:eqIndex])
 			body = part[eqIndex+1:]
+
 		case len(part) > 1 && part[0] == '"' && part[len(part)-1] == '"':
 			name = "text"
 			body = part
-		case isQuotedText(part):
+
+		case len(part) > 1 && part[0] == '\'' && part[len(part)-1] == '\'':
 			name = "text"
 			body = part
-		case reXPathSelector.Match([]byte(part)) || strings.HasPrefix(part, ".."):
-			// If selector starts with '//' or '//' prefixed with multiple opening
-			// parenthesis, consider xpath. @see https://github.com/microsoft/playwright/issues/817
-			// If selector starts with '..', consider xpath as well.
+
+		case reXPathSelector.MatchString(part) || strings.HasPrefix(part, ".."):
 			name = "xpath"
 			body = part
-		case strings.HasPrefix(part, "internal:role="):
-			name = "internal:role"
-			body = part
-		case strings.HasPrefix(part, "internal:attr="):
-			name = "internal:attr"
-			body = part
-		case strings.HasPrefix(part, "internal:label="):
-			name = "internal:label"
-			body = part
-		case strings.HasPrefix(part, "internal:text="):
-			name = "internal:text"
-			body = part
-		case strings.HasPrefix(part, "internal:control="):
-			name = "internal:control"
-			body = part[eqIndex+1:]
+
 		default:
 			name = "css"
 			body = part
 		}
 
 		capture := false
-		if name[0] == '*' {
+		if strings.HasPrefix(name, "*") {
 			capture = true
 			name = name[1:]
 		}
 
 		return &SelectorPart{Name: name, Body: body}, capture
 	}
 
-	if !strings.Contains(s.Selector, ">>") {
-		part, capture := parsePart(s.Selector, 0, len(s.Selector))
-		err := s.appendPart(part, capture)
-		if err != nil {
-			return err
+	start := 0
+	index := 0
+	var quote rune
+	selector := s.Selector
```

Comment:
Please use a `var` block to keep variables tidy:

```suggestion
	var (
		start, index int
		quote rune
		selector = s.Selector
	)
```

Do we need a `selector` variable since we have `s.Selector` already?